### DISCUSSION TITLE
Problem with MPRester in pymatgen 2022.9.21

### DISCUSSION THREAD

**Discussion by Arindamsannyal** [materialsproject/pymatgen#3001] [Q&A]
I have installed `pymatgen` v. 2022.9.21 on my Linux system and I have the MP API key. However, when I tried to use the MPRester to get data from the Materials Project, it provides several 'AttributeError:'s. For instance for a simple script as below, only the `m.get_structure_by_material_id` works. 

```python
from pymatgen.ext.matproj import MPRester
with MPRester("apikey") as m:

    # Structure for material id
    structure = m.get_structure_by_material_id("mp-1234")
    print(structure)

    data = m.get_data("Fe2O3")
    print(data)

    # Dos for material id
    dos = m.get_dos_by_material_id("mp-1234")

    # Bandstructure for material id
    bandstructure = m.get_bandstructure_by_material_id("mp-1234")

    # Get all energies of materials with formula "*2O".
    results = m.query("*2O", ['energy'])

    # Get the formulas and energies of materials with materials_id mp-1234
    # or with formula FeO.
    results = m.query("FeO mp-1234", ['pretty_formula', 'energy'])

    # Get all compounds of the form ABO3
    results = m.query("**O3", ['pretty_formula', 'energy'])
```

For other results, pymatgen gives the 'AttributeError:'s. For example, AttributeError: '_MPResterNew' object has no attribute 'query'. 

Can you please suggest the possible reasons for these errors and how to solve them?

![AttributeError](https://user-images.githubusercontent.com/33083005/195759606-8f770c9c-86f1-4211-9c82-1d1274af47de.JPG)

**Comment by munrojm** [comment]
`mpr.query` has been replaced with `mpr.summary.search`. You can use that new method and pass all of your wildcard formulas to the `formula` argument.
